Feature: Discard nil value field (#425)
procore-oss/blueprinter: README.md

</details>

<details>

> <summary>Discarding nil fields</summary>

Reviewer: **suggestion**: Slight nitpick, but I'd suggest framing this as "omit" or "exclude" instead of "discard". 

```suggestion
<summary Exclude Fields with nil Values</summary>
```

> By default nil values are included. You can override this behavior by passing `:discard_nil` option.

Reviewer: ```suggestion
By default, fields with `nil` values are included when rendering. You can override this behavior by setting `:exclude_if_nil: true` in the field definition. 
```

> Usage:
> 
> ```ruby
> class UserBlueprint < Blueprinter::Base
>   identifier :uuid
> 
>   field :name
>   field(:birthday, discard_nil: true) do |_user, options|
>     nil
>   end
> end

Reviewer: Let's simplify this a bit:

```suggestion
Usage:

```ruby
class UserBlueprint < Blueprinter::Base
  identifier :uuid

  field :name
  field :birthday, exclude_if_nil: true
end

user = User.new(name: 'John Doe')
puts UserBlueprint.render(user)
```
